### PR TITLE
feature/basic_layout

### PR DESCRIPTION
**Pull request by aringlot**
refactoring according to review summary
translate

### CODE REVIEW

File: app/app.config.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-function routeConfig($stateProvider) {
+function config($stateProvider, $translateProvider, $translatePartialLoaderProvider) {
```
Comment (gskachkov) [inline app/app.config.js]: I also can be arrow function, it is better to use `ngInject` in this case. You forger to add translatePartialLoaderProvider within .$inject

File: app/app.config.js [hunk 2]
```diff
@@ -8,7 +8,15 @@ function routeConfig($stateProvider) {
...
+        })
+        .preferredLanguage('en')
+        .useLoaderCache(true)
+        .useCookieStorage();
 }
 
-routeConfig.$inject = ['$stateProvider'];
-export default routeConfig;
\ No newline at end of file
+config.$inject = ['$stateProvider', '$translateProvider'];
```
Comment (gskachkov) [inline app/app.config.js]: Different number of parameters in function and in inject

File: app/app.module.js [hunk 3]
```diff
@@ -1,17 +1,24 @@
...
-var app = angular.module('registratorApp', ['ui.router', 'common', 'aboutUs', 'home']);
-app.config(config);
-app.run([
-    '$state',
-     function($state){
-        $state.go('main');
-     }
- ]);
+angular
+    .module('registratorApp', ['ui.router','pascalprecht.translate', 'ngCookies', 'common', 'aboutUs', 'home'])
```
Comment (gskachkov) [inline app/app.module.js]: import can return name of the module
so use can do in following manner:
```
import uiRouter from 'angular-ui-router';
...
angular.module('registratorApp', [uiRouter ... ])

```

File: app/app.module.js [hunk 4]
```diff
@@ -1,17 +1,24 @@
...
+    .config(config)
+    .run(($state, $rootScope) => {
+            'ngInject'
+            $state.go('main');
+
+            $rootScope.$on('$translatePartialLoaderStructureChanged', function () {
```
Comment (gskachkov) [inline app/app.module.js]: Can this be replaced by ()=>$translate.refresh() ?

File: app/modules/common/components/button.component.js [hunk 5]
```diff
@@ -1,12 +1,15 @@
...
         name: '<',
         url: '<',
         customClass: '<'
     },
-    templateUrl: './views/common/button.template.html',
-    controller: 'ButtonController',
-    controllerAs: 'buttonCtrl'
+    template: template,
+    controller: Controller,
+    controllerAs: 'VM'
```
Comment (gskachkov) [inline app/modules/common/components/button.component.js]: please use lowercase for vm

**Comment by aringlot** [comment]
points, that were mentioned after review, are mostly fixed except bower -> npm

**Pull request merged**